comp/core/tagger/tagger.go

```diff
@@ -324,11 +327,48 @@ func (t *TaggerClient) ResetCaptureTagger() {
 // NOTE(remy): it is not needed to sort/dedup the tags anymore since after the
 // enrichment, the metric and its tags is sent to the context key generator, which
 // is taking care of deduping the tags while generating the context key.
-func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, udsOrigin string, clientOrigin string, cardinalityName string) {
-	cardinality := taggerCardinality(cardinalityName)
+func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, originInfo taggertypes.OriginInfo) {
```

Comment:
Just a small question, I think performance was a concern for `EnrichTag`. If the input is not a pointer then the struct is copied to the function. Given that the struct has many fields in means we might do many copies. Are we fine with this ?

Reply:
The benchmark is still good 👍